### PR TITLE
Statemachine and model added. Jupyter notebook examples created

### PR DESCRIPTION
**Pull request by andrewbheyer**

### CODE REVIEW

File: python/lsst/ts/scheduler/__init__.py [hunk 1]
```diff
@@ -7,3 +7,5 @@
 from .driver import *
 from .main import *
 from .sal_utils import *
+from .model import *
+from .scheduler_csc import *
```
Comment (tribeiro) [inline python/lsst/ts/scheduler/__init__.py]: Add new line at the end of the file...

File: python/lsst/ts/scheduler/scheduler_csc.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        self.disable = salpylib.DDSController(self.context, 'disable')
+        self.standby = salpylib.DDSController(self.context, 'standby')
+        self.exit_control = salpylib.DDSController(self.context, 'exitControl')
+
+        self.enter_control.start()
+        self.start.start()
+        self.enable.start()
+        self.disable.start()
+        self.standby.start()
+        self.exit_control.start()
```
Comment (tribeiro) [inline python/lsst/ts/scheduler/scheduler_csc.py]: Add new line at the end of the file...

File: python/lsst/ts/scheduler/stateMachine/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+from .target_producer import *
```
Comment (tribeiro) [inline python/lsst/ts/scheduler/stateMachine/__init__.py]: Add new line at the end of the file...

File: python/lsst/ts/scheduler/stateMachine/states.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    def exit(self, model):
+        # Exiting Enable, pause the thread, we don't know if we want to kill yet
+        self.target_producer.pause()
+
+
+class FaultState(csc_states.FaultState):
+
+    def __init__(self, subsystem_tag):
+        super(FaultState, self).__init__(subsystem_tag)
```
Comment (tribeiro) [inline python/lsst/ts/scheduler/stateMachine/states.py]: Add new line at the end of the file...

File: python/lsst/ts/scheduler/stateMachine/target_producer.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        self.log_switch = True
+        # TODO: Calls on the model telling it to save whatever needs saving
+
+    def produce_targets(self):
+        # Threading module only allows start() to every be called once
+        if self.isAlive():
+            self.start_flag.set()
+        else:
+            self.start()
+            self.start_flag.set()
```
Comment (tribeiro) [inline python/lsst/ts/scheduler/stateMachine/target_producer.py]: Add new line at the end of the file...

File: python/lsst/ts/scheduler/stateMachine/target_producer.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
+import time
+import threading
+import logging
+
+from SALPY_scheduler import *
```
Comment (tribeiro) [inline python/lsst/ts/scheduler/stateMachine/target_producer.py]: I don't think it is a good idea to `import *`. Specially, note that you are only using a couple of the definitions here, `SAL_scheduler` and `scheduler_logevent_targetC` so you can easily import them individually.

File: python/lsst/ts/scheduler/model.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import time
+from SALPY_scheduler import *
+
+DEFAULT_STATE = "OFFLINE"
+
+class Model():
+	"""The model contains all business logic related to the scheduler.
```
Comment (tribeiro) [inline python/lsst/ts/scheduler/model.py]: This file uses tab as indentation. Need to fix it to comply with the other files.

File: python/lsst/ts/scheduler/model.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
+import time
+from SALPY_scheduler import *
```
Comment (tribeiro) [inline python/lsst/ts/scheduler/model.py]: I don't think it is a good idea to `import *`. Here you are using only `SAL_scheduler` and `scheduler_logevent_validSettingsC()`, so its better to import them individually.

File: python/lsst/ts/scheduler/__init__.py (hunk 1 above)
Comment (andrewbheyer) [reply python/lsst/ts/scheduler/__init__.py]: done

File: python/lsst/ts/scheduler/scheduler_csc.py (hunk 2 above)
Comment (andrewbheyer) [reply python/lsst/ts/scheduler/scheduler_csc.py]: done

File: python/lsst/ts/scheduler/stateMachine/__init__.py (hunk 3 above)
Comment (andrewbheyer) [reply python/lsst/ts/scheduler/stateMachine/__init__.py]: done

File: python/lsst/ts/scheduler/stateMachine/states.py (hunk 4 above)
Comment (andrewbheyer) [reply python/lsst/ts/scheduler/stateMachine/states.py]: done

File: python/lsst/ts/scheduler/stateMachine/target_producer.py (hunk 5 above)
Comment (andrewbheyer) [reply python/lsst/ts/scheduler/stateMachine/target_producer.py]: done

File: python/lsst/ts/scheduler/stateMachine/target_producer.py (hunk 6 above)
Comment (andrewbheyer) [reply python/lsst/ts/scheduler/stateMachine/target_producer.py]: good point, done

File: python/lsst/ts/scheduler/model.py (hunk 7 above)
Comment (andrewbheyer) [reply python/lsst/ts/scheduler/model.py]: hmmm I am wondering how this is happening, I thought it was my remote editor that did that but I fixed it... regardless, done

File: python/lsst/ts/scheduler/model.py (hunk 8 above)
Comment (andrewbheyer) [reply python/lsst/ts/scheduler/model.py]: good point, done

**Pull request merged**